components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java

```diff
@@ -1692,6 +1697,21 @@ public void updateLocalSp(HttpServletRequest request) {
         serviceProvider.setApplicationName(request.getParameter("spName"));
         serviceProvider.setDescription(request.getParameter("sp-description"));
 
+        // Check if isSystemDefaultApp property is already set.
+        boolean isSystemDefaultAppPropExist = false;
+        for (ServiceProviderProperty property : serviceProvider.getSpProperties()) {
```

Comment:
Check if getSpProperties can be null

Reply:
SP properties are initialized when creating the ServiceProvider object. So no need to check for null. ref: https://github.com/wso2/carbon-identity-framework/blob/eece0de2e5793bdcc43658b81434df71bb1788f5/components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java#L101